src/makeInfinityAdder.js

```diff
@@ -6,7 +6,21 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  return function adder(number) {
+    if (number === undefined) {
+      const totalAmount = sum;
+
+      sum = 0;
+
+      return totalAmount;
+    } else {
+      sum += number;
+
+      return adder;
+    }
```

Comment:
you don't need `else` here, anyways you return value from previous `if`
```suggestion
    }
    
    sum += number;

    return adder;
```